File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        s, e = 0, 1
        maxProfit = 0

        while e < len(prices):
            maxProfit = max(maxProfit , prices[e] - prices[s])
            if prices[s] > prices[e]:
                s = e
            e += 1
        return maxProfit



        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.java
class Solution {
    public int characterReplacement(String s, int k) {
        char[] charS = s.toCharArray();
        Map<Character, Integer> map = new HashMap<>();

        int start = 0;
        int maxLen = 0;
        
        for (int end=0; end < charS.length; end++){
            map.put(charS[end], map.getOrDefault(charS[end], 0) + 1);

            while (((end-start+1)-Collections.max(map.values())) > k) {
                map.put(charS[start], map.get(charS[start]) -1);
                start++;
            }

            maxLen = Math.max(maxLen, end-start+1);
        }

        return maxLen;
    }
}
